File: backend/src/QuickBite.Shared/ApiErrorResponse.cs
namespace QuickBite.Shared;

public sealed class ApiErrorResponse
{
    public DateTime Timestamp { get; init; } = DateTime.UtcNow;
    public int Status { get; init; }
    public string Error { get; init; } = string.Empty;
    public string Message { get; init; } = string.Empty;
    public string Path { get; init; } = string.Empty;
    public Dictionary<string, string[]>? Details { get; init; }
}

File: mobile/lib/src/core/error/api_error_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'api_error_response.g.dart';

@JsonSerializable()
class ApiErrorResponse {
  const ApiErrorResponse({
    this.timestamp,
    this.status,
    this.error,
    this.message,
    this.path,
    this.details,
  });

  factory ApiErrorResponse.fromJson(Map<String, dynamic> json) =>
      _$ApiErrorResponseFromJson(json);
  Map<String, dynamic> toJson() => _$ApiErrorResponseToJson(this);

  final DateTime? timestamp;
  final int? status;
  final String? error;
  final String? message;
  final String? path;
  final Map<String, List<String>>? details;
}
